src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,293 @@
+        });
+    }
+
+    private setEventsFromMxObjects = (mxObjects: mendix.lib.MxObject[]) => {
+        const { titleAttribute, startAttribute, endAttribute } = this.props;
+        const events = mxObjects.map(mxObject => {
+            const title = mxObject.get(titleAttribute) as string;
+            const start = new Date(mxObject.get(startAttribute) as number);
+            const end = new Date(mxObject.get(endAttribute) as number);
+            const guid = mxObject.getGuid();
+
+            return {
+                title,
+                start,
+                end,
+                guid
+            };
+        });
+        this.setState({ events });
+    }
+
+    private fetchEventsByMicroflow(mxObject: mendix.lib.MxObject) {
+        const { dataSourceMicroflow } = this.props;
+        if (mxObject && dataSourceMicroflow) {
+            window.mx.ui.action(dataSourceMicroflow, {
+                callback: (mxObjects: mendix.lib.MxObject[]) => this.setEventsFromMxObjects(mxObjects),
+                error: error => window.mx.ui.error(`Error while executing microflow: ${dataSourceMicroflow}: ${error.message}`),
+                params: {
+                    applyto: "selection",
+                    guids: mxObject ? [ mxObject.getGuid() ] : []
+                }
+            });
+        }
+    }
+
+    private onClickEvent = (eventInfo: any) => {
+        mx.data.get({
+            guid: eventInfo.guid,
+            callback: (object) => this.excecuteEventAction(object),
+            error: error => window.mx.ui.error(`Error while executing action: ${error.message}`)
+        });
+    }
+
+    private excecuteEventAction = (object: mendix.lib.MxObject) => {
+        const { onClickEvent, eventMicroflow, mxform, eventNanoflow, openPageAs, eventPage } = this.props;
+        if (!object || !object.getGuid()) {
+            return;
+        }
+        const context = new mendix.lib.MxContext();
+        context.setContext(object.getEntity(), object.getGuid());
+        if (onClickEvent === "callMicroflow" && eventMicroflow) {
+            window.mx.ui.action(eventMicroflow, {
+                error: error => window.mx.ui.error(`Error while executing microflow: ${eventMicroflow}: ${error.message}`),
+                origin: mxform,
+                context
+            });
+        } else if (onClickEvent === "callNanoflow" && eventNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                context,
+                error: error => window.mx.ui.error(
+                    `An error occurred while executing the nanoflow: ${error.message}`
+                ),
+                nanoflow: eventNanoflow,
+                origin: mxform
+            });
+        } else if (onClickEvent === "showPage" && eventPage) {
+            window.mx.ui.openForm(eventPage, {
+                context,
+                error: error => window.mx.ui.error(
+                    `Error while opening page ${eventPage}: ${error.message}`
```

Comment:
Think this can fit on the previous line.

Reply:
Done